src/soc/ibm/power9/istep_8_1.c

```diff
@@ -0,0 +1,337 @@
+/* SPDX-License-Identifier: GPL-2.0-only */
+
+#include <arch/io.h>
+#include <stdint.h>
+#include <string.h>
+#include <console/console.h>
+#include <cpu/power/istep_8.h>
+#include <cpu/power/mvpd.h>
+#include <cpu/power/scom.h>
+
+#include "fsi.h"
+#include "homer.h"
+#include "scratch.h"
+
+/*
+ * Legend for constant names:
+ * - *_FSI is a CFAM address (in 4 byte words, that's how it is in Hostboot)
+ * - *_FSI_BYTE is an FSI address
+ */
+
+/* Used to read SBE Boot Side from processor */
+const uint64_t SBE_BOOT_SELECT_MASK = 0x0000400000000000;
+
+/* Builds bitmask of functional cores based on Partial Good vector stored in PG
+ * keyword of CP00 record in CPU-specific MVPD partition */
+static uint64_t get_available_cores(uint8_t chip)
+{
+	enum {
+		VPD_CP00_PG_HDR_LENGTH   = 1,
+		VPD_CP00_PG_DATA_LENGTH  = 128,
+		VPD_CP00_PG_DATA_ENTRIES = VPD_CP00_PG_DATA_LENGTH / 2,
+
+		ALL_ON_PG_MASK = 0xFFFF,
+		EC_AG_MASK     = 0xE1FF,
+	};
+
+	uint64_t cores = 0;
```

Comment:
If this function is made reusable, this should be made static table to avoid reading MVPD multiple times. Doesn't have to be in this PR, though.

Reply:
In changes related to HOMER I moved it to `mvpd.c` and made public (it's still used only for one CPU and called only once in romstage and once in ramstage).